fe/fe-core/src/main/java/com/starrocks/connector/hive/CachingHiveMetastore.java

```diff
@@ -411,6 +452,13 @@ private List<HivePartitionName> getPresentPartitionNames(LoadingCache<HivePartit
                 .collect(Collectors.toList());
     }
 
+    private Map<HivePartitionName, Partition> getPresentPartitions(LoadingCache<HivePartitionName, Partition> cache,
+                                                             String dbName, String tableName) {
+        return cache.asMap().entrySet().stream()
```

Comment:
This will iterate and compare over all cache entries, I'm afraid it will introduce a lot of CPU overhead for comparison. Please use `getAllPresent` instead.